chapter02/worksheets/solution2.2.6.1.scala

```diff
@@ -0,0 +1,28 @@
+/** 
+Exercise 2.2.6.1 Implement a function fromPairs that performs the inverse transformation to the toPairs function defined in Example 2.2.5.6. The required type
+signature and a sample test are:
+
+def fromPairs[A](xs: Seq[(A, A)]): Seq[A] = ???
+
+scala> fromPairs(List((a,b), (c,<nothing>)))
+res1: Seq[(String, String)] = List("a", "b", "c", "<nothing>")
+
+Hint: This can be done with foldLeft or with flatMap.
+ */
+
+def fromPairs[A](xs: Seq[(A, A)]): Seq[A] = { 
+  var b: Seq[A] = List()
+  xs.map((x: (A, A)) => (b = b :+ (x: (A, A))(0) :+ (x: (A, A))(1)))
```

Comment:
The function for extracting the parts from the tuple are `_1` and `_2`. The syntax `x(0)` will not work with tuples until Scala 3.

Also, it is much easier to use `flatMap` or `flatten` for implementing this function, instead of using `var b` and mutating `b` in a loop.

Reply:
Okay, thanks. Should provide the alternative solutions too.

Reply:
Would prefer that, as comments are easier to read then. But do, as it's convenient for you.